shellcheck_gha/extractor.py

```diff
@@ -93,11 +105,22 @@ def run(self) -> tuple[int, int, list[Finding]]:
 
         logger.info("Scanning the directory: %s", self.directory)
         for yaml_path in self.iter_workflow_paths():
-            num_files_scanned += 1
             logger.info("Checking the YAML file: %s", yaml_path)
 
             with yaml_path.open() as fp:
-                parsed_yaml = GitHubYAML.model_validate(yaml.safe_load(fp))
+                try:
+                    parsed_yaml = GitHubYAML.model_validate(yaml.safe_load(fp))
+                except ValidationError as exc:
+                    # Skip the file if we are not in a strict mode.
+                    if self.raise_on_unsupported_yaml is False:
+                        logger.warning(
+                            "Skipping invalid YAML file (%s): %s", yaml_path, exc
```

Comment:
The file by itself isn't _invalid_ but unsupported (invalid sounds like there is a syntax error), it may prevent some confusions and also keeps consistency with the variable name `raise_on_unsupported_yaml`:
```suggestion
                            "Skipping unsupported YAML file (%s): %s", yaml_path, exc
```